new/language/implementation/java/java.go

```diff
@@ -0,0 +1,308 @@
+				if scopedNode := scope.Lookup(node.Content()); scopedNode != nil {
+					node.UnifyWith(scopedNode)
+				}
+			}
+
+			// user.getName(name, lastName)
+			if parent.Type() == "arguments_list" {
+				scope.Assign(node.Content(), node)
+			}
+			// todo: see what this is
+			// case "property_identifier":
+			// 	parent := node.Parent()
+			// 	if parent != nil && slice.Contains(variableLookupParents, parent.Type()) {
+			// 		if scopedNode := scope.Lookup(node.Content()); scopedNode != nil {
+			// 			node.UnifyWith(scopedNode)
+			// 		}
+			// 	}
+			// }
+		}
+		return visitChildren()
+	})
+}
+
+// TODO: See if anything needs to be added here
+func (implementation *javaImplementation) ExtractPatternVariables(input string) (string, []patternquerytypes.Variable, error) {
+	nameIndex := patternQueryVariableRegex.SubexpIndex("name")
+	typesIndex := patternQueryVariableRegex.SubexpIndex("types")
+	i := 0
+
+	var params []patternquerytypes.Variable
+
+	replaced, err := regex.ReplaceAllWithSubmatches(patternQueryVariableRegex, input, func(submatches []string) (string, error) {
+		nodeTypes := strings.Split(submatches[typesIndex], "|")
+		if nodeTypes[0] == "" {
+			nodeTypes = []string{"_"}
+		}
+
+		for _, nodeType := range nodeTypes {
+			if !slices.Contains(allowedPatternQueryTypes, nodeType) {
+				return "", fmt.Errorf("invalid node type '%s' in pattern query", nodeType)
+			}
+		}
+
+		dummyValue := produceDummyValue(i, nodeTypes[0])
+
+		params = append(params, patternquerytypes.Variable{
+			Name:       submatches[nameIndex],
+			NodeTypes:  nodeTypes,
+			DummyValue: dummyValue,
+		})
+
+		i += 1
+
+		return dummyValue, nil
+	})
+
+	if err != nil {
+		return "", nil, err
+	}
+
+	return replaced, params, nil
+}
+
+func produceDummyValue(i int, nodeType string) string {
+	return "CurioVar" + fmt.Sprint(i)
+}
+
+// TODO: See if anything needs to be added here
+func (implementation *javaImplementation) AnonymousPatternNodeParentTypes() []string {
+	return anonymousPatternNodeParentTypes
+}
+
+// TODO: See if anything needs to be added here
+func (implementation *javaImplementation) FindPatternMatchNode(input []byte) [][]int {
+	return matchNodeRegex.FindAllIndex(input, -1)
+}
+
+// TODO: See if anything needs to be added here
+func (implementation *javaImplementation) FindPatternUnanchoredPoints(input []byte) [][]int {
+	return ellipsisRegex.FindAllIndex(input, -1)
+}
+
+func (implementation *javaImplementation) PatternMatchNodeContainerTypes() []string {
+	return patternMatchNodeContainerTypes
+}
+
+func (javaImplementation *javaImplementation) ShouldSkipNode(node *tree.Node) bool {
+	return false
+}
+
+func (*javaImplementation) PatternLeafContentTypes() []string {
+	return []string{
+		// todo: see if type identifier should be removed from here (User user) `User` is type
+		// todo see if `modifier` should be here it's a single token containing `string public final static`
```

Comment:
- Identifier is good I think
- Yes I'd say add `modifier`, though we probably won't use it in practice